libminifi/include/SchedulingAgent.h

```diff
@@ -190,11 +207,27 @@ class SchedulingAgent {
   std::shared_ptr<core::controller::ControllerServiceProvider> controller_service_provider_;
 
  private:
+  struct SchedulingInfo {
+    std::chrono::time_point<std::chrono::steady_clock> start_time_ = std::chrono::steady_clock::now();
+    mutable std::chrono::time_point<std::chrono::steady_clock> last_alert_time_ = std::chrono::steady_clock::now();
```

Comment:
Don't lie on the interface about `const`-ness, unless dealing with mutexes, caches or similar.

Reply:
I don't think it's a lie. 
It think it depends on preference, in my opinion marking some helper members mutable and maintaining const correctness is better than having a lot of non-const functions that look like they should have been.

Reply:
Before doing some additional research, my approach was: I should have a very good reason to use `mutable`. The ability to mark `watchDogFunc` as `const` is not a good enough reason IMO, but this requires subjective judgement.

Now I came across a great approach on stackoverflow:
> [mutable] allows the differentiation of bitwise const and logical const. Logical const is when an object doesn't change in a way that is visible through the public interface, like [mutex].
https://stackoverflow.com/a/105061/3997716

The change of `last_alert_time_` is observable from the public interface which makes marking it `mutable` objectively wrong. It can even be changed through the public interface.

Reply:
Tried to change it, but it cannot be changed as the object is used in a set and only const iteration can be performed on that. Added a comment to indicate the reason of being mutable.

Reply:
ok, thanks